src/baldr/datetime.cc

```diff
@@ -55,6 +55,10 @@ tz_db_t::tz_db_t() : db(date::get_tzdb()) {
   for (const auto& zone : db.zones) {
     names.emplace(zone.name(), ++idx);
   }
+  // add the aliases of the links here with the index of their targets
+  for (const auto& link : db.links) {
+    names.emplace(link.name(), names.find(link.target())->second);
+  }
```

Comment:
this should be all we need. previously valhalla_build_timezones would alias old timezone names from the shapefile in the sqlite db. that adds a fair bit of burden. `date_time` has its own list of those "links" where the `name()` is the old tz name and the `target()` is the new alias, equivalent to what we were doing ourselves before. now the whole aliasing bound to the `date_time` library, which is so much nicer. and no one has to wonder about those aliases anymore.

Reply:
I've merged changes from master into our micro branch and noticed that this change breaks US_OS_TZDB build option. It won't build because there is no links member inside tzdb, when US_OS_TZDB=1 is defined.

quote from data/tz.h
```
struct tzdb
{
    std::string                 version = "unknown";
    std::vector<time_zone>      zones;
#if !USE_OS_TZDB
    std::vector<time_zone_link> links;
#endif
```

Is it expected limitation?

Reply:
iirc we never wanted to use the os tzdb as the ordering of stuff could change and therefore make data that is incompatible with our ordering of the tz names.

Reply:
I've seen in some other sources this define and thought it's supported.

Reply:
ok. I'll refactor our part to use embed tzdb from third_parties. Thanks for clarification.

Reply:
it'd be nice to be able to use system tzdbs. and I kinda figured we're able to do just that, now that we're hard-coding things even more.

> It won't build because there is no links member inside tzdb, when US_OS_TZDB=1 is defined.

that's a pity and I wonder how we'd get aliases in system dbs.

anyways, if anyone ever badly wants it, they can dive in:)